src/pretty/ListUtil.re

```diff
@@ -0,0 +1,16 @@
+let split_first = (xs: list('a)): ('a, list('a)) =>
+  switch (xs) {
+  | [] => failwith("empty list")
+  | [first, ...trailing] => (first, trailing)
+  };
+
+let split_last_opt = (xs: list('a)): option((list('a), 'a)) =>
+  switch (List.rev(xs)) {
+  | [] => None
+  | [y, ...ys] => Some((List.rev(ys), y))
+  };
+let split_last = (xs: list('a)): (list('a), 'a) =>
+  switch (split_last_opt(xs)) {
+  | None => failwith("empty list")
+  | Some(r) => r
+  };
```

Comment:
Duplicated some code here for now, created an issue to dedup https://github.com/hazelgrove/hazel/issues/424